tls/s2n_key_update.c

```diff
@@ -146,3 +146,12 @@ int s2n_check_record_limit(struct s2n_connection *conn, struct s2n_blob *sequenc
 
     return S2N_SUCCESS;
 }
+
+int s2n_connection_request_key_update(struct s2n_connection *conn, bool peer_update_requested)
+{
+    POSIX_ENSURE_REF(conn);
+    /* s2n-tls does not currently support requesting key updates from peers */
+    POSIX_ENSURE(!peer_update_requested, S2N_ERR_T_USAGE);
+    s2n_atomic_flag_set(&conn->key_update_pending);
```

Comment:
Maybe we should have another check to make sure that the handshake is complete?

Reply:
Theoretically not necessary? We just wouldn't send the update until after the handshake

Reply:
Well it'd be nice to be able to have the compliance comment about not sending this until the finished message has been sent, so I'll go ahead and add it.

Reply:
But this method doesn't send anything, so that'd be the wrong place for that compliance comment

Reply:
Yup, I'd add it as a test case in `s2n_key_update_test.c`